common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1874,6 +1874,16 @@ def _update_item_from_fields(
                 block_data.fields = settings
 
                 new_id = new_structure['_id']
+
+                # Stuck in published mode
+                # Source_version for draft block is present when it is from pubslihed structure at revert_to_published()
+                # after this source_verion is not updated for draft block but its update_version always updated
+                # as result has_changes() return false (first look in source_version and then update_version)
+                # by changing the source_version to None has_changes() will pick update_version instead source_version
+                # and prevent/eliminate at stuck on publish mode state
```

Comment:
How about using this comment instead? (if it's accurate...)
```
# When a draft block from a published structure is reverted in revert_to_published(),
# the update_version is updated while source_version is not. If source_version is not None,
# set it to None - so that has_changes() will use update_version for comparison and return True.
# Prevents the block from getting stuck in the published state.
```